kolibri/plugins/setup_wizard/assets/src/views/index.vue

```diff
@@ -42,33 +83,82 @@
   module.exports = {
     $trNameSpace: 'setupWizard',
     $trs: {
-      header: 'Create device owner and facility',
-      deviceOwner: 'Device Owner',
+      formHeader: 'Create device owner and facility',
+      deviceOwnerSectionHeader: 'Device Owner',
+      facilitySectionHeader: 'Facility',
+      usernameInputLabel: 'Username',
+      passwordInputLabel: 'Password',
+      confirmPasswordInputLabel: 'Confirm password',
+      facilityInputLabel: 'Facility name',
       deviceOwnerDescription: 'To use Kolibri, you first need to create a Device Owner. This account will be used to configure high-level settings for this installation, and create other administrator accounts',
-      username: 'Username',
-      password: 'Password',
-      confirmPassword: 'Confirm password',
-      facility: 'Facility',
       facilityDescription: 'You also need to create a Facility. This represents your school, training center, or other installation location',
-      facilityName: 'Facility name',
-      getStarted: 'Create and get started',
+      formSubmissionButton: 'Create and get started',
+      usernameFieldEmptyErrorMessage: 'Username cannot be empty',
+      usernameCharacterErrorMessage: 'Username cannot contain a "."',
+      passwordFieldEmptyErrorMessage: 'Password cannot be empty',
+      passwordsMismatchErrorMessage: 'Passwords do not match',
+      facilityFieldEmptyErrorMessage: 'Facility cannot be empty',
+      cannotSubmitPageError: 'Please resolve all of the errors shown',
+      genericPageError: 'Something went wrong',
     },
     data() {
       return {
         username: '',
-        username_error: false,
+        usernameError: null,
         password: '',
-        confirm_password: '',
-        password_error: false,
+        passwordConfirm: '',
+        passwordError: null,
         facility: '',
-        facility_error: false,
-        errormessage: '',
+        facilityError: null,
+        globalError: null,
       };
     },
+    components: {
+      'core-textbox': require('kolibri.coreVue.components.textbox'),
+      'icon-button': require('kolibri.coreVue.components.iconButton'),
+      'ui-alert': require('keen-ui/src/UiAlert'),
+    },
+    computed: {
+      firstUsernameFieldVisit() {
+        return this.usernameError === null;
+      },
+      usernameFieldPopulated() {
+        return !!this.username;
+      },
+      usernameValidityCheck() {
```

Comment:
The username has to be alphanum and max 30 chars. https://github.com/learningequality/kolibri/blob/develop/kolibri/auth/models.py#L190

I added this check to the user-create-modal.
https://github.com/learningequality/kolibri/blob/develop/kolibri/plugins/management/assets/src/views/user-page/user-create-modal.vue#L124